src/main/scala/org/broadinstitute/dsde/agora/server/webservice/routes/MockAgoraDirectives.scala

```diff
@@ -15,9 +14,15 @@ trait MockAgoraDirectives extends AgoraDirectives {
     provide(AgoraConfig.mockAuthenticatedUserEmail)
   }
 
+  // allow unit tests to specify the user making the request; if they don't, use the default
   def usernameFromRequest(magnet: ImplicitMagnet[ExecutionContext]): Directive1[String] = {
-    provide(AgoraConfig.mockAuthenticatedUserEmail)
+    optionalHeaderValueByName(MockAgoraDirectives.mockUserHeader) map {
+      case Some(mockuser) => mockuser
```

Comment:
Kinda pedantic, but this method switches between "user", "user name", and "user email"; I assume these all refer to email?